plog-client/src/main/java/com/airbnb/plog/client/PlogClient.java

```diff
@@ -0,0 +1,137 @@
+package com.airbnb.plog.client;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+import com.google.common.hash.Hashing;
+
+/**
+ * ## Plog client for Java
+ * Example:
+ *
+ * ```java
+ * PlogClient plogClient = new PlogClient("127.0.0.1", 23456, 64000);
+ * plogClient.send("My hovercraft is full of eels.");
+ * ```
+ * You can configure the client at initialization by passing these options:
+ *
+ * + host - The host of the Plog process (e.g., 'localhost')
+ * + port - The port on which Plog is listening (e.g., 23456)
+ * + chunkSize - The maximum payload size for multipart datagrams (e.g., 64,000)
+ */
+@Slf4j
+public class PlogClient {
+
+  private AtomicInteger lastMessageId;
+
+  private final InetAddress address;
+
+  private final int port;
+
+  private final int chunkSize;
+
+  private DatagramSocket socket;
+
+  public PlogClient(String host, int port, int chunkSize) {
```

Comment:
nevermind will add a constant and another constructor with no chunkSize required